src/test/java/org/commonjava/util/partyline/JoinFileWriteTwiceWithDelayTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.util.partyline;
+
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+@RunWith( org.jboss.byteman.contrib.bmunit.BMUnitRunner.class )
+@BMUnitConfig( loadDirectory = "target/test-classes/bmunit", debug = true )
+public class JoinFileWriteTwiceWithDelayTest
+        extends AbstractJointedIOTest
+{
+
+    @Test
+    @BMScript( "JoinFileWriteTwiceWithDelay.btm" )
+    public void joinFileWriteTwiceWithDelay()
+            throws Exception
+    {
+        final CountDownLatch latch = new CountDownLatch( 3 );
+        final JoinableFile stream = startTimedWrite( 1, latch );
+        startRead( 0, stream, latch );
+        startRead( 500, stream, latch );
```

Comment:
For something like this, you might be able to use a byteman timer to delay the second one's joinStream() call.